okhttp/src/main/java/okhttp3/internal/Util.java

```diff
@@ -498,6 +499,15 @@ public static int checkDuration(String name, long duration, TimeUnit unit) {
     return (int) millis;
   }
 
+  public static int checkDuration(String name, Duration duration) {
+    if (duration == null) throw new NullPointerException("duration == null.");
+    long millis = duration.toMillis();
+    if (millis < 0) throw new IllegalArgumentException(name + " < 0");
+    if (millis > Integer.MAX_VALUE) throw new IllegalArgumentException(name + " too large.");
+    if (millis == 0) throw new IllegalArgumentException(name + " too small.");
```

Comment:
Doesn't this conflict with the javadocs for 0?

Reply:
no conflict :)

Reply:
> A value of 0 means no timeout, otherwise values must be between 1 and {@link Integer#MAX_VALUE} when converted to milliseconds.

Reply:
Yes, if call `Duration.ofXXX(0)` method, I think request won't success because of limited Broadband. So` throw new IllegalArgumentException(name + " too small.");`